### PR TITLE
Flush by flush channels

### PR DESCRIPTION
**Pull request by bigsheeper**
issue: https://github.com/milvus-io/milvus/issues/26820

/kind bug

### CODE REVIEW

**Comment by bigsheeper** [comment]
rerun ut

**Comment by xiaofan-luan** [comment]
please fix the flush message

File: internal/types/types.go [hunk 1]
```diff
@@ -301,7 +304,7 @@ type DataCoord interface {
 	// WatchChannels notifies DataCoord to watch vchannels of a collection
 	WatchChannels(ctx context.Context, req *datapb.WatchChannelsRequest) (*datapb.WatchChannelsResponse, error)
 	// GetFlushState gets the flush state of multiple segments
```
Comment (XuanYang-cn) [inline internal/types/types.go:306]: need to update this comment

File: internal/datacoord/services.go [hunk 2]
```diff
@@ -116,24 +117,46 @@ func (s *Server) Flush(ctx context.Context, req *datapb.FlushRequest) (*datapb.F
...
+		return resp, nil
+	}
+
 	log.Info("flush response with segments",
 		zap.Int64("collectionID", req.GetCollectionID()),
 		zap.Int64s("sealSegments", sealedSegmentIDs),
 		zap.Int64s("flushSegments", flushSegmentIDs),
-		zap.Time("timeOfSeal", timeOfSeal))
+		zap.Time("timeOfSeal", timeOfSeal),
+		zap.Uint64("flushTs", ts))
```
Comment (XuanYang-cn) [inline internal/datacoord/services.go:152]: perhaps transfer ts into human readable time.Time for readability

File: internal/datanode/channel_meta.go [hunk 3]
```diff
@@ -147,10 +159,12 @@ func newChannel(channelName string, collID UniqueID, schema *schemapb.Collection
 
 		segments: make(map[UniqueID]*Segment),
 
-		needToSync: atomic.NewBool(false),
+		isHighMemory: atomic.NewBool(false),
+		flushTs:      atomic.NewUint64(math.MaxUint64),
```
Comment (XuanYang-cn) [inline internal/datanode/channel_meta.go:163]: why new segments have the maxUint64 as flushTs, why not 0

File: internal/datanode/channel_meta.go [hunk 4]
```diff
@@ -114,7 +118,15 @@ type ChannelMeta struct {
...
 
-	needToSync   *atomic.Bool
+	// isHighMemory is intended to trigger the syncing of segments
+	// when segment's buffer consumes a significant amount of memory.
+	isHighMemory *atomic.Bool
+
+	// flushTs is intended to trigger:
+	// 1. the syncing of segments when consumed ts exceeds flushTs;
+	// 2. the updating of channelCP when channelCP exceeds flushTs.
+	flushTs *atomic.Uint64
```
Comment (XuanYang-cn) [inline internal/datanode/channel_meta.go:128]: So its a minTs for channel to flush all streaming data <= flushTs into S3, which means to trigger a sync when bufferred segments ts >= flushTs, and update cp if buffered channelCP.ts reaches flustTs, took me a while to understands the meaning of the param.

Perhaps rename it with syncThreshold, means any segments whose buffer exceeds the thresholds with be synced

and set a const NoThreshold = math.MaxInt64, cuz math.MaxInt64 do make people wonder why when reading codes

Comment (XuanYang-cn) [reply internal/datanode/channel_meta.go:128]: setSyncThreshold(ts)

cleanSyncThreahold/resetSyncThreshold or ....

would be betther to replace setFlushts

**Review by XuanYang-cn — CHANGES_REQUESTED** [review]
Over all LGTM, just mentioned some readbility suggestions

File: internal/datacoord/services.go [hunk 5]
```diff
@@ -116,24 +117,46 @@ func (s *Server) Flush(ctx context.Context, req *datapb.FlushRequest) (*datapb.F
...
+			})
+			err = s.cluster.FlushChannels(ctx, nodeID, ts, channelNames)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		resp.Status = merr.Status(err)
```
Comment (xiaofan-luan) [inline internal/datacoord/services.go:143]: log the error?

**Review by xiaofan-luan — COMMENTED** [review]
/approve

**Comment by bigsheeper** [comment]
rerun ut

**Pull request merged**